src/main/java/htsjdk/samtools/util/AbstractProgressLogger.java

```diff
@@ -44,35 +46,51 @@ protected AbstractProgressLogger(final String noun, final String verb, final int
      */
     abstract protected void log(String ... message);
 
+    private synchronized void log(final String chrom, final int pos) {
+        final long now = System.currentTimeMillis();
+        final long lastPeriodSeconds = (now - this.lastStartTime) / 1000;
+        this.lastStartTime = now;
 
+        final long seconds = (now - startTime) / 1000;
+        final String elapsed   = formatElapseTime(seconds);
+        final String period    = pad(fmt.format(lastPeriodSeconds), 4);
+        final String processed = pad(fmt.format(this.processed), 13);
 
-    @Override
-    public synchronized boolean record(final String chrom, final int pos) {
-	    if (this.lastStartTime == -1) this.lastStartTime = System.currentTimeMillis();
-	    if (++this.processed % this.n == 0) {
-            final long now = System.currentTimeMillis();
-            final long lastPeriodSeconds = (now - this.lastStartTime) / 1000;
-            this.lastStartTime = now;
+        final String readInfo;
+        if (chrom == null) readInfo = "*/*";
+        else readInfo = chrom + ":" + fmt.format(pos);
 
-            final long seconds = (System.currentTimeMillis() - startTime) / 1000;
-            final String elapsed   = formatElapseTime(seconds);
-            final String period    = pad(fmt.format(lastPeriodSeconds), 4);
-            final String processed = pad(fmt.format(this.processed), 13);
-
-            final String readInfo;
-            if (chrom == null) readInfo = "*/*";
-            else readInfo = chrom + ":" + fmt.format(pos);
+        log(this.verb, " ", processed, " " + noun + ".  Elapsed time: ", elapsed, "s.  Time for last ", fmt.format(this.n),
+                ": ", period, "s.  Last read position: ", readInfo);
+    }
 
-            log(this.verb, " ", processed, " " + noun + ".  Elapsed time: ", elapsed, "s.  Time for last ", fmt.format(this.n),
-                    ": ", period, "s.  Last read position: ", readInfo);
+    /**
+     * Logs the last last record if it wasn't previoiusly logged.
+     * @return boolean true if logging was triggered, false otherwise
+     */
+    public synchronized boolean recordLast() {
```

Comment:
I think this should probably just be called `log()` since it's not recording anything and technically could be used any time you want to print progress, not just at the last record.